tests/bgp/test_bgp_authentication.py

```diff
@@ -0,0 +1,258 @@
+'''This script is to test the EBGP Authentication feature of SONiC.
+'''
+import logging
+
+import pytest
+import time
+
+from tests.common.helpers.constants import DEFAULT_NAMESPACE
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+bgp_config_sleeptime = 60
+bgp_pass = "sonic.123"
+mismatch_pass = "badpassword"
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, enum_asic_index):
+    duthost = duthosts[rand_one_dut_hostname]
+    dut_asn = tbinfo['topo']['properties']['configuration_properties']['common']['dut_asn']
+
+    dut_lldp_table = duthost.shell("show lldp table")['stdout'].split("\n")[3].split()
+    tor1 = dut_lldp_table[1]
```

Comment:
@Azarack current implementation cannot guarantee you are choosing T0.

e.g.
admin@masic:~$ show lldp table
Capability codes: (R) Router, (B) Bridge, (O) Other
LocalPort    RemoteDevice      RemotePortID    Capability    RemotePortDescr
-----------  ----------------  --------------  ------------  -----------------
Ethernet0    **T2**        Ethernet1       BR

you may loop through output of: `duthost.shell("show lldp table")['stdout'].split("\n")` and find if 'T0' in it

Reply:
@wenyiz2021 It is not important what type of neighbor is chosen.  I no longer have a T1 instance available so the changes were done on T2 and the neighbor can be either T1 or T3.

Reply:
@Azarack 
I have tried the command on multi-asic T1, your current implementation depends on order of the neighbors shown by `show lldp table`, for T1 it can be either tor or T2 for `dut_lldp_table[1]`, is this what you expected?

Reply:
@wenyiz2021 Yes, that is what I expect.  I'm looking for a BGP neighbor relationship to test functionality and it doesn't matter what layer the neighbor is in.